### PR TITLE
feat: add refresh_oauth_token() API

### PR DESCRIPTION
**Pull request by Verdgil**
request for access_token is exported to new method

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+ # Change import
+
+* fix init imports
+
+## [1.2.2](https://github.com/casdoor/casdoor-python-sdk/compare/v1.2.1...v1.2.2) (2022-12-02) 
+
```
Comment (hsluoyz) [inline CHANGELOG.md:6]: Why change this?

File: src/casdoor/main.py [hunk 2]
```diff
@@ -15,6 +15,9 @@
 import requests
 import jwt
 import json
+
```
Comment (hsluoyz) [inline src/casdoor/main.py:18]: Use existing `import requests`

File: src/tests/test_oauth.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+from requests import Response
```
Comment (hsluoyz) [inline src/tests/test_oauth.py:14]: new line

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+ # Change import
```
Comment (hsluoyz) [inline CHANGELOG.md:1]: Can we avoid modifying this file?

File: src/casdoor/main.py [hunk 5]
```diff
@@ -70,14 +71,51 @@ def get_oauth_token(self, code: str) -> str:
         :param code: the code that sent from Casdoor using redirect url back to your server.
         :return: access_token
         """
+        r = self.oauth_token_request(code)
+        access_token = r.json().get("access_token")
+
+        return access_token
+
+    def oauth_token_request(self, code: str) -> requests.Response:
```
Comment (hsluoyz) [inline src/casdoor/main.py:79]: Don't import breaking change

File: src/casdoor/main.py [hunk 6]
```diff
@@ -15,6 +15,7 @@
 import requests
 import jwt
 import json
+
```
Comment (hsluoyz) [inline src/casdoor/main.py:18]: Don't change it

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Verdgil if you are adding new API, add it to casdoor-go-sdk first

File: src/casdoor/main.py (hunk 5 above)
Comment (Verdgil) [reply src/casdoor/main.py:78]: > Don't import breaking change

Can your explain where is "breaking change"?

Comment (hsluoyz) [reply src/casdoor/main.py:78]: No problem now

File: src/casdoor/main.py [hunk 7]
```diff
@@ -70,14 +70,51 @@ def get_oauth_token(self, code: str) -> str:
...
         params = {
             "grant_type": self.grant_type,
             "client_id": self.client_id,
             "client_secret": self.client_secret,
             "code": code,
         }
-        r = requests.post(url, params)
+        return requests.post(url, params)
+
+    def refresh_token_request(self, refresh_token: str, scope: str = "") -> requests.Response:
```
Comment (hsluoyz) [inline src/casdoor/main.py:93]: Add the new API to README

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: src/casdoor/main.py [hunk 8]
```diff
@@ -70,14 +70,51 @@ def get_oauth_token(self, code: str) -> str:
...
+        params = {
+            "grant_type": self.grant_type,
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+            "scope": scope,
+            "refresh_token": refresh_token,
+        }
+        return requests.post(url, params)
+
+    def get_oauth_refreshed_token(self, refresh_token: str, scope: str = "") -> str:
```
Comment (hsluoyz) [inline src/casdoor/main.py:110]: Rename to `refresh_oauth_token` based on Go's naming: https://github.com/casdoor/casdoor-go-sdk/blob/35bd7afc4c75b7465557c53bb783201b30dcfb02/casdoorsdk/token.go#L53

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Verdgil see: https://github.com/casdoor/casdoor-python-sdk/pull/32#discussion_r1041734446

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@Verdgil also add this new API to README.md

image

**Pull request merged**